### PR TITLE
Fixes #1625. Fixes checks on Guild Dungeon Investment and separates Guild Dungeon mapflags for pre-renewal and renewal

### LINKED ISSUE #1625: Guild Dungeon Investment
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Issue Prelude

[//]: # (Please complete these mandatory steps and check the following boxes by putting an `x` inside the brackets _before_ filing your issue)

-  [x] I have not modified the source prior to reproducing this issue.
-  [x] I am using the latest version of Hercules.
-  [x] I am aware that this report will be closed or deleted if it becomes obvious that I am stating the false.

Issues related to Guild Dungeon Investment and Hall of the Abyss Dungeons

[//]: # (Description of the problem or issue at length.)
[//]: # (Please specify any battle configuration related to the components of this issue that have been changed from the default values. This will allow quicker determination of the cause of the problem.)

### Current Behavior

- When calculating funds checks if funds on npc [#fund_master] are more than or equal to 7000 and sets .@total_7000 in that case. However, script ends before open guild dungeon because the check that opens it is never reached.
- When buying Thanks_Invest_Ticket, the npcs who call the function checks if Weight < 1000, but player can buy up to 200 tickets and every ticket has 10 weight, so, when buying more than 100 tickets can lose some of them if don't have enough space to receive it.
- First guild dungeons should remove gvg_dungeon and set gvg_dungeon mapflag to new hall of the abyss dungeons. This needs to divide mapflags for pre-renewal and renewal.

### Expected Behavior

- Grant access to Hall of the Abyss when funds are reached.
- Do proper check when player buys Thanks_Invest_Ticket.
- Have updated mapflags.

**Branch(es):** `master`

**Hercules rev. hash/commit:** 

Git revision src: d0568eefba012ec1339f474154a5f239d08f4b41

### PR DESCRIPTION
**Pull request by iRagno**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

- Fixes to issues on checks of Guild Dungeon Investment.
- Separates mapflag gvg_dungeon from common to pre-renewal and renewal.
-- pre-renewal: set gvg_dungeon to woe fe guild dungeons
-- renewal: sets gvg_dungeon to hall of the abyss dungeons (removes gvg_dungeon to woe fe guild dungeons).
- Update some numbers to constants.

**Affected Branches:** `master`

[//]: # (Master? Slave?)

**Issues addressed:** #1625 Guild Dungeon Investment

### Known Issues and TODO List

None

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

---
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1627)

### CODE REVIEW

File: npc/re/woe-fe/invest_main.txt [hunk 1]
```diff
@@ -1520,67 +1556,126 @@ malangdo,215,119,4	script	Thanks Ticket Machine	2_VENDING_MACHINE1,{
...
+		else if (.@i <= 1140)
+			callsub(L_Bag, Unripe_Fruit, 5);
+		else if (.@i <= 1150)
+			callsub(L_Bag, Dried_Yggdrasilberry, 5);
+		else if (.@i <= 1250)
+			callsub(L_Bag, Sg_White_Potion_Box, 1);
+		else if (.@i <= 1471)
+			callsub(L_Bag, Old_Blue_Box, 1);
+		else if (.@i <= 1571)
+			callsub(L_Bag, Poison_Bottle_Box2, 1, "Poison Bottle Box(30);");
```
Comment (Ridley8819) [inline npc/re/woe-fe/invest_main.txt]: missing " " (starting from here)

File: npc/re/woe-fe/invest_main.txt [hunk 2]
```diff
@@ -1589,75 +1684,142 @@ malangdo,215,119,4	script	Thanks Ticket Machine	2_VENDING_MACHINE1,{
...
+		else if (.@i <= 830)
+			callsub(L_Bag, Unripe_Fruit, 10);
+		else if (.@i <= 840)
+			callsub(L_Bag, Dried_Yggdrasilberry, 10);
+		else if (.@i <= 940)
+			callsub(L_Bag, Sg_White_Potion_Box, 1);
+		else if (.@i <= 1140)
+			callsub(L_Bag, Old_Blue_Box, 1);
+		else if (.@i <= 1240)
+			callsub(L_Bag, Poison_Bottle_Box2, 1, Poison Bottle Box(30));
```
Comment (Ridley8819) [inline npc/re/woe-fe/invest_main.txt]: missing " " (starting from here)

File: npc/re/woe-fe/invest_main.txt [hunk 3]
```diff
@@ -1589,75 +1684,142 @@ malangdo,215,119,4	script	Thanks Ticket Machine	2_VENDING_MACHINE1,{
...
+		else if (.@i <= 1340)
+			callsub(L_Bag, Water_Of_Darkness, 20);
+		else if (.@i <= 1440)
+			callsub(L_Bag, Box_Of_Grudge, 10);
+		else if (.@i <= 1540)
+			callsub(L_Bag, Sleepy_Box, 10);
+		else if (.@i <= 1690)
+			callsub(L_Bag, Box_Of_Sunlight, 4);
+		else if (.@i <= 1790)
+			callsub(L_Bag, Elemental_Fire, 10, Elemental Converter[Fire]);
```
Comment (Ridley8819) [inline npc/re/woe-fe/invest_main.txt]: missing " "

File: npc/re/woe-fe/invest_main.txt [hunk 4]
```diff
@@ -1589,75 +1684,142 @@ malangdo,215,119,4	script	Thanks Ticket Machine	2_VENDING_MACHINE1,{
...
+		else if (.@i <= 1890)
+			callsub(L_Bag, Elemental_Water, 10, Elemental Converter[Water]);
```
Comment (Ridley8819) [inline npc/re/woe-fe/invest_main.txt]: missing " "

File: npc/re/woe-fe/invest_main.txt [hunk 5]
```diff
@@ -1589,75 +1684,142 @@ malangdo,215,119,4	script	Thanks Ticket Machine	2_VENDING_MACHINE1,{
...
+		else if (.@i <= 1990)
+			callsub(L_Bag, Elemental_Wind, 10, Elemental Converter[Wind]);
```
Comment (Ridley8819) [inline npc/re/woe-fe/invest_main.txt]: missing " "

File: npc/re/woe-fe/invest_main.txt [hunk 6]
```diff
@@ -1589,75 +1684,142 @@ malangdo,215,119,4	script	Thanks Ticket Machine	2_VENDING_MACHINE1,{
...
+		else if (.@i <= 2090)
+			callsub(L_Bag, Elemental_Earth, 10, Elemental Converter[Earth]);
```
Comment (Ridley8819) [inline npc/re/woe-fe/invest_main.txt]: missing " "

**Comment by iRagno** [comment]
I'm sorry for that, I did it semi automatized on the last ones and the process remove quotation marks. Sorry.

I have fixed it. Thanks for the observation @Ridley8819.

**Comment by Mysteries** [comment]
Thoughts @MishimaHaruna

**Comment by Ridley8819** [comment]
i would have merged it, bu I am not sure regarding pre-re here :)

**Comment by iRagno** [comment]
Mapflags for First Floor of Guild Dungeons can be easily checked on official server with Guild Investment Dungeon, since it opens Guild Dungeons to anyone, even to those players who doesn't belong to any guild.

**Comment by iRagno** [comment]
I just check it on iRO and Guild Dungeons (classic old ones) does not have gvg_dungeon mapflag, meaning players can't attack each other. I have recorded the test, please tell me if needed replay.

**Comment by Ridley8819** [comment]
It would be great if you can attach it here! Thanks for your efforts.

**Comment by iRagno** [comment]
This has been recorded today on iRO Chaos (renewal). In this test I put two characters into gld_dun03 to test if they would be able to receive damage from each others. The characters I use were a Ranger and a Magician, only the magician has guild (no other was logged in that momment and also guild has not any alliance). I can't share the name of the Ranger or any data from him.

http://www.mediafire.com/file/hfab7c1xdlqynal/mapflag_test.zip

In the test, players can't attack each other on gld_dun03, and confirms guild dungeon doesn't have any more the gvg_dungeon mapflag.

**Pull request merged**